runtime/near-vm-runner/src/wasmtime_runner.rs

```diff
@@ -42,16 +42,22 @@ fn with_caller<T>(func: impl FnOnce(&mut Caller) -> T) -> T {
 }
 
 impl MemoryLike for WasmtimeMemory {
-    fn fits_memory(&self, offset: u64, len: u64) -> Result<(), ()> {
-        let end = offset.checked_add(len).ok_or(())?;
-        let end = usize::try_from(end).map_err(|_| ())?;
+    fn fits_memory(&self, slice: MemSlice) -> Result<(), ()> {
+        let end = slice.end::<usize>()?;
         if end <= with_caller(|caller| self.0.data_size(caller)) {
             Ok(())
         } else {
             Err(())
         }
     }
 
+    fn view_memory(&self, slice: MemSlice) -> Result<Cow<[u8]>, ()> {
+        let range = slice.range::<usize>()?;
+        with_caller(|caller| {
+            self.0.data(caller).get(range).map(|slice| Cow::Owned(slice.to_vec())).ok_or(())
```

Comment:
Is this making a copy here because the lifetime of the returned slice is tied to the `caller`?

Reply:
Yes.